OneCommunityGlobal/HGNRest
# Ajay- fix person name search on Project Page

# Description
This PR adds a backend endpoint that provides autocomplete suggestions for person names when entered. It returns the first name, last name, and ID from the database.
Fixes # (bug list priority medium 47)

## Related PRS (if any):
To test this backend PR you need to checkout the #[2676](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2676) frontend PR.
…

## Main changes explained:
- Update userProfileController.js to add the logic for handling autocomplete functionality, retrieving first name, last name, and ID from the database.
- Update userProfileRouter.js to define the route for the new autocomplete endpoint, linking it to the controller. 
…

## How to test:
1. Check into the current branch
2. Do `npm install` , `npm run build` and `npm start` to run this PR locally
3. Open Postman to access to the endpoint
4. First create a POST request to `http://localhost:4500/api/login` using the owner credentials in the body to get the token.
5.  <img width="1907" alt="token" src="https://github.com/user-attachments/assets/2e14df45-2287-48cb-8055-99dd2001466a">
6. Copy the token and create a new GET request `http://localhost:4500/api/userProfile/autocomplete/name` replacing the name with the text you want to search along with the token in the authorization
7. <img width="1898" alt="get_request" src="https://github.com/user-attachments/assets/e19e456b-a4a6-41ab-a7fe-0ef23c15fef7">
8. The request will return the user's suggested First Name, Last Name, and ID.
